test/es-module/test-esm-import-flag.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+require('../common');
+const fixtures = require('../common/fixtures');
+const assert = require('assert');
+const { spawnSync } = require('child_process');
+
+const cjsEntry = fixtures.path('es-modules', 'cjs-file.cjs');
+const cjsImport = fixtures.fileURL('es-modules', 'cjs-file.cjs');
+const mjsEntry = fixtures.path('es-modules', 'mjs-file.mjs');
+const mjsImport = fixtures.fileURL('es-modules', 'mjs-file.mjs');
+
+{
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import', mjsImport,
+      '-e', 'console.log("log")',
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.match(stdout, /\.mjs file\r?\nlog/);
+}
+
+{
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import', mjsImport,
+      '--input-type', 'module',
+      '-e', 'console.log("log")',
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.match(stdout, /\.mjs file\r?\nlog/);
+}
+
+{
+  // Import esm when main entrypoint is a cjs file
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import', mjsImport,
+      cjsEntry,
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.match(stdout, /\.mjs file\r?\n\.cjs file/);
+}
+
+{
+  // Import cjs when main entrypoint is a mjs file
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import', cjsImport,
+      mjsEntry,
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.match(stdout, /\.cjs file\r?\n\.mjs file/);
+}
+
+
+{
+  // Import esm when main entrypoint is a cjs file
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import', mjsImport,
+      cjsEntry,
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.match(stdout, /\.mjs file\r?\n\.cjs file/);
+}
+
+{
+  // Import same file twice
+  const { status, stderr, stdout } = spawnSync(
+    process.execPath,
+    [
+      '--import', mjsImport,
+      '--import', mjsImport,
+      cjsEntry,
+    ],
+    { encoding: 'utf8' },
+  );
+
+  assert.strictEqual(status, 0);
+  assert.strictEqual(stderr, '');
+  assert.match(stdout, /\.mjs file\r?\n\.cjs file/);
```

Comment:
Can we test how many time the code of `mjsImport` gets executed? Currently this test would pas whether it's executed once or twice. We could use a global variable that increment itself, or validate the full stdout.